orderbook/src/database/events.rs

```diff
@@ -48,17 +48,21 @@ impl Database {
                 (SELECT COALESCE(MAX(block_number), 0) FROM trades), \
                 (SELECT COALESCE(MAX(block_number), 0) FROM settlements), \
                 (SELECT COALESCE(MAX(block_number), 0) FROM invalidations));";
-        let block_number: i64 = sqlx::query_scalar(QUERY)
+        let start = Instant::now();
```

Comment:
It's unfortunate that sqlx doesn't provide any way to add these metrics in one place (at least I couldn't find any).

Perhaps we can add a RAII object to avoid manually measuring time? Just a thought.

Reply:
I think the rust prometheus library does provide a RAII guard or we could implement our own. We have already been measuring time manually like this in a few places so I did it the same way. But I am definitely open to improving it everywhere.

Reply:
There is https://docs.rs/prometheus/0.12.0/prometheus/struct.HistogramTimer.html

IIrc, I started measuring time manually for the API requests, because we needed the `Reply` in order to know which labels we want the metric to be on (cf [here](https://github.com/gnosis/oba-services/blob/ece6ab9cc9226772a3671bdacf9af220aef95d24/orderbook/src/metrics.rs#L122)). I couldn't find a way to set labels after starting the timer so I just took the start instant manually.

Reply:
I wonder if this could be worked around by wrapping the queries in a `timed_query` and only having to start and stop the timer in one place...

In python, this could then be invoked by adding a "decorator" to each of the queries, I wonder if there is something similar here...

```

fn timed_query(query, message_string) {
    start_time
    inner_query(query)
    stop_time
    log_metrics(message_string)
}

@timed_query(message_string = "some metric table name")
fn inner_query(query) {}
```